atlasaction/action_test.go

```diff
@@ -0,0 +1,159 @@
+package atlasaction
+
+import (
+	"bytes"
+	"context"
+	"database/sql"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"ariga.io/atlas-go-sdk/atlasexec"
+	_ "github.com/mattn/go-sqlite3"
+	"github.com/sethvargo/go-githubactions"
+	"github.com/stretchr/testify/require"
+)
+
+func TestMigrateApply(t *testing.T) {
+	t.Run("local dir", func(t *testing.T) {
+		tt := newT(t)
+		tt.setInput("url", "sqlite://"+tt.db)
+		tt.setInput("dir", "file://testdata/migrations/")
+
+		err := MigrateApply(context.Background(), tt.cli, tt.act)
+		require.NoError(t, err)
+
+		m, err := tt.outputs()
+		require.EqualValues(t, map[string]string{
+			"error": "",
+		}, m)
+	})
+	t.Run("tx-mode", func(t *testing.T) {
+		tt := newT(t)
+		tt.setInput("url", "sqlite://"+tt.db)
+		tt.setInput("dir", "file://testdata/migrations/")
+		tt.setInput("tx-mode", "fake")
+		err := MigrateApply(context.Background(), tt.cli, tt.act)
+
+		// An error here proves that the tx-mode was passed to atlasexec, which
```

Comment:
I thought we were aiming for tests that use an HTTP server and check that calls were made?

Also, relying on Atlas raw error strings seem fragile to me. These error strings could easily be changed in the Atlas repository and no one would think that this affects this repository

Reply:
This test checks a local flow, cloud mode is added in one of the next PRs which will use a local HTTP server.

re the tests fragility (I prefer the Google-y term "brittle", I agree it's not iron-clad. However, there is a trade-off, not relying on any error message (only the exit code) doesn't prove the argument is passed. Relying on mocking on the other hand is less brittle, but has its own costs and risks which can be discussed.

In this case, the risk of the test breaking because of an upstream error message change is low and has no impact on the user, so I opt for the simple way even if it is not 100% textbook (In general, I try to be practical and not religious about "The Right Way™")